File: src/fr/gstraymond/android/CardPagerActivity.java
package fr.gstraymond.android;

import static fr.gstraymond.constants.Consts.POSITION;
import android.app.Fragment;
import android.os.Bundle;
import fr.gstraymond.R;

public class CardPagerActivity extends CardCommonActivy {

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_card_pager);
		
		Fragment fragment = new CardPagerFragment();
		fragment.setArguments(getBundle());
		replaceFragment(fragment, R.id.card_pager_container);
	}

	@Override
	protected Bundle getBundle() {
		Bundle bundle = super.getBundle();
		bundle.putInt(POSITION, getPosition());
		return bundle;
	}
	
	private int getPosition() {
		return getIntent().getIntExtra(POSITION, 0);
	}
}

File: app/src/main/kotlin/fr/gstraymond/android/CardPagerActivity.kt
package fr.gstraymond.android

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.fragment.app.Fragment
import fr.gstraymond.R
import fr.gstraymond.android.fragment.CardPagerFragment
import fr.gstraymond.models.search.response.Card
import fr.gstraymond.utils.find

class CardPagerActivity : CardCommonActivity(R.layout.activity_card_pager) {

    companion object {
        const val POSITION_EXTRA = "position"

        fun getIntent(context: Context, card: Card) =
                Intent(context, CardPagerActivity::class.java).apply {
                    putExtra(CARD_EXTRA, card)
                }

        fun getIntent(context: Context, card: Card, position: Int) =
                getIntent(context, card).apply {
                    putExtra(POSITION_EXTRA, position)
                }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setSupportActionBar(find(R.id.toolbar))
        true.actionBarSetDisplayHomeAsUpEnabled()

        replaceFragment(CardPagerFragment(), R.id.card_pager_container, getBundle())
    }

    override fun getBundle() = super.getBundle().apply {
        putInt(POSITION_EXTRA, intent.getIntExtra(POSITION_EXTRA, 0))
    }

    override fun replaceFragment(fragment: Fragment, id: Int) {
        if (supportFragmentManager.findFragmentById(id) == null) {
            super.replaceFragment(fragment, id)
        }
    }

    override fun onSaveInstanceState(outState: Bundle) {
        //super.onSaveInstanceState(outState);
    }
}
